src/org/zaproxy/zap/extension/ascanrulesAlpha/ChallengeCallbackAPI.java

```diff
@@ -17,49 +17,67 @@
  */
 package org.zaproxy.zap.extension.ascanrulesAlpha;
 
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
 import java.util.TreeMap;
+
+import org.apache.commons.httpclient.URIException;
 import org.apache.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.db.DatabaseException;
+import org.parosproxy.paros.model.HistoryReference;
 import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
 import org.parosproxy.paros.network.HttpMessage;
 import org.zaproxy.zap.extension.api.ApiException;
-import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.callback.CallbackImplementor;
+import org.zaproxy.zap.extension.callback.ExtensionCallback;
 
 /**
  * General Abstract class for Challenge/Response Active Plugin management
- * 
+ *
  * @author yhawke (2014)
  */
-public abstract class ChallengeCallbackAPI extends ApiImplementor {
+public abstract class ChallengeCallbackAPI implements CallbackImplementor {
 
-    // This are the result contents that should be returned by the API
-    private static final String API_RESPONSE_KO = "ko";
-    private static final String API_RESPONSE_OK = "ok";
-    
     // The default expiration time for each callback (in millisecs)
     private static final long CALLBACK_EXPIRE_TIME = 2 * 60 * 1000;
-    
+
```

Comment:
I understand wanting things to look nice (I was guilty of including all kinds of white spaces changes in my original PRs). However, mixing formatting changes with code changes makes review harder.

https://github.com/zaproxy/zaproxy/wiki/DevGuidelines#formatting

Reply:
In this case I made the formatting changes knowingly, because I copied the class from here https://github.com/zaproxy/zap-extensions/blob/beta/src/org/zaproxy/zap/extension/ascanrulesBeta/ChallengeCallbackAPI.java

The changes should be compared between
https://github.com/zaproxy/zap-extensions/blob/beta/src/org/zaproxy/zap/extension/ascanrulesBeta/ChallengeCallbackAPI.java
and
https://github.com/denniskniep/zap-extensions/blob/56191c323e0de66511c2da870318c7fbedee9256/src/org/zaproxy/zap/extension/ascanrulesAlpha/ChallengeCallbackAPI.java

I think it would make it easier to include the changes later in the beta branch.
What do you think?

Reply:
Hmmm weird. I wonder why github isn't treating it as a new file if it's a different path.

Reply:
The class already exists in the alpha branch. I simply updated it. Copied the changes from the class in the beta branch to the class in the alpha branch.

Reply:
Oh ok, that makes sense.